apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ExtensionContext.java

```diff
@@ -49,11 +73,23 @@ void deserialize(String value) {
         if (StringUtil.isEmpty(value)) {
             return;
         }
-        final String[] extensionParts = value.split("-");
+        String[] extensionParts = value.split(SEPARATOR);
```

Comment:
I think you misunderstand me, I mean no value should be represented by ` `, the separator should always be `-`.